### PR TITLE
add new API torch.amp.is_autocast_available

### PR DESCRIPTION
**Pull request by guangyey**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #124938

# Motivation
expose `torch._is_autocast_available` to `torch.amp.is_autocast_available` as a public api.

cc @mcarilli @ptrblck @leslie-fang-intel @jgong5

### CODE REVIEW

File: torch/__init__.py [hunk 1]
```diff
@@ -1477,7 +1477,7 @@ def manager_path():
         raise RuntimeError("Unable to find torch_shm_manager at " + path)
     return path.encode('utf-8')
 
-from torch.amp import autocast, GradScaler
+from torch.amp import autocast, GradScaler, is_autocast_available
```
[the remark this answers is not in the document]
Comment (guangyey) [reply torch/__init__.py]: Got it.

File: torch/amp/autocast_mode.py [hunk 2]
```diff
@@ -6,7 +6,11 @@
 import torch
 from torch.types import _dtype
 
-__all__ = ["autocast_decorator", "autocast"]
+__all__ = ["autocast_decorator", "autocast", "is_autocast_available"]
+
+
+def is_autocast_available(device_type: str):
```
[the remark this answers is not in the document]
Comment (guangyey) [reply torch/amp/autocast_mode.py]: Add a doc.

**Review by albanD — APPROVED** [review]
SGTM once CI is good!

**Pull request closed without merging**